web/src/tracking/activation.tsx

```diff
@@ -0,0 +1,153 @@
+        map(dataAndErrors => {
+            if (!dataAndErrors.data) {
+                return null
+            }
+            const data = dataAndErrors.data
+            if (!data.repositories.nodes) {
+                return null
+            }
+            const rURLs = data.repositories.nodes
+                .filter(r => r.gitRefs && r.gitRefs.totalCount > 0)
+                .sort(r => (r.gitRefs ? -r.gitRefs.totalCount : 0))
+                .map(r => r.url)
+            if (rURLs.length === 0) {
+                return null
+            }
+            return rURLs[0]
+        })
+    )
+
+const fetchActivationStatus = (isSiteAdmin: boolean) => () =>
+    queryGraphQL(
+        isSiteAdmin
+            ? gql`
+                  query {
+                      externalServices {
+                          totalCount
+                      }
+                      repositories(enabled: true) {
+                          totalCount
+                      }
+                      viewerSettings {
+                          final
+                      }
+                      users {
+                          totalCount
+                      }
+                      currentUser {
+                          usageStatistics {
+                              searchQueries
+                              findRefsActions
+                          }
+                      }
+                  }
+              `
+            : gql`
+                  query {
+                      currentUser {
+                          usageStatistics {
+                              searchQueries
+                              findRefsActions
+                          }
+                      }
+                  }
+              `
+    ).pipe(
+        map(dataOrThrowErrors),
+        map(data => {
+            const authProviders = window.context.authProviders
+            const completed: { [key: string]: boolean } = {
+                didSearch: !!data.currentUser && data.currentUser.usageStatistics.searchQueries > 0,
+                'action:findReferences': !!data.currentUser && data.currentUser.usageStatistics.findRefsActions > 0,
+            }
+            if (isSiteAdmin) {
+                completed.connectedCodeHost = data.externalServices && data.externalServices.totalCount > 0
+                completed.enabledRepository =
+                    data.repositories && data.repositories.totalCount !== null && data.repositories.totalCount > 0
+                if (authProviders) {
+                    completed.enabledSignOn =
+                        data.users.totalCount > 1 || authProviders.filter(p => !p.isBuiltin).length > 0
+                }
+            }
+            return completed
+        })
+    )
+
+/**
+ * Returns a new ActivationStatus instance to use in the web app.
+ *
+ * @param isSiteAdmin determines if site-admin-only activation steps are included.
+ */
+export const newActivationStatus = (isSiteAdmin: boolean) => {
```

Comment:
Factory functions in TS/JS are named `create*` because functions should start with an imperative verb.
```suggestion
export const createActivationStatus = (isSiteAdmin: boolean) => {
```